vmm/src/device_manager.rs

```diff
@@ -1017,6 +1009,39 @@ impl DeviceManager {
         Ok(device_manager)
     }
 
+    pub fn start_serial_manager_thread(
+        &self,
+    ) -> DeviceManagerResult<thread::JoinHandle<result::Result<(), SerialManagerError>>> {
+        // caller should check if a serial device exists
+        let serial = self.console.serial.as_ref().unwrap().clone();
+        let serial_config = self.config.lock().unwrap().serial.clone();
+        let serial_tty = serial_config.mode == ConsoleOutputMode::Tty;
+        let serial_pty = if self.serial_pty.is_some() {
+            Some(
+                self.serial_pty
+                    .as_ref()
+                    .unwrap()
+                    .lock()
+                    .unwrap()
+                    .main
+                    .try_clone()
+                    .unwrap(),
+            )
+        } else {
+            None
+        };
+
+        let thread = thread::Builder::new()
+            .name("serial-manager".to_string())
+            .spawn(move || {
```

Comment:
You want to use the `exit_evt` to write to when the thread panics (using `std::panic::catch_unwind`)

Reply:
Ah okay, I am reusing the `DeviceManager`'s `exit_evt` as it didn't seem to make much sense to have a different one given the shutdown probably would still be the way to respond to the evt write.